infrastructure/src/main/java/org/corfudb/infrastructure/BatchProcessor.java

```diff
@@ -105,7 +123,20 @@ private void processor() {
 
                     if (currOp == null || processed == BATCH_SIZE
                             || currOp == BatchWriterOperation.SHUTDOWN) {
-                        streamLog.sync(sync);
+                        Runnable fsyncRunnable = () -> {
+                            try {
+                                streamLog.sync(sync);
+                            } catch (IOException e) {
+                                throw new RuntimeException(e);
+                            }
+                        };
+                        if (fsyncTimer.isPresent()) {
+                            fsyncTimer.get().record(fsyncRunnable);
```

Comment:
This is not the fsync latency, but rather the fsync time of all dirty segments. Based on the current batch + segment size at most it the sum of fsyncing two files.

Reply:
Fixed